src/EIP712.sol

```diff
@@ -59,13 +77,19 @@ contract EIP712 is IEIP712, IERC5267 {
     /// @notice Returns the `domainSeparator` used to create EIP-712 compliant hashes.
     /// @return The 32 bytes domain separator result.
     function domainSeparator() public view returns (bytes32) {
+        return _domainSeparator();
+    }
+
+    /// @notice Internal domainSeparator implementation for use in ERC7739
+    function _domainSeparator() internal view returns (bytes32) {
         return
             keccak256(abi.encode(_DOMAIN_TYPEHASH, _cachedNameHash, _cachedVersionHash, block.chainid, address(this)));
     }
 
-    /// @notice Public getter for `_hashTypedData()` to produce a replay-safe hash from the given `hash`.
-    /// @param hash The nested typed data hash as defined by EIP-712. Assumes the hash is the result of applying EIP-712 hashStruct.
-    /// @return The corresponding replay-safe hash.
+    /// @notice Public getter for `_hashTypedData()` to produce a EIP-712 hash using this account's domain separator
+    /// @dev This is meant to be used for internal verification of SignedBatchedCalls and thus does not produce signatures that are replay-safe.
+    ///      See ERC7739 for a version that is replay-safe and can be used for external verification through ERC-1271.
+    /// @param hash The nested typed data. Assumes the hash is the result of applying EIP-712 hashStruct.
     function hashTypedData(bytes32 hash) public view virtual returns (bytes32) {
```

Comment:
yeah same here, it feels like there is no need to define an internal function and a public function that do the same thing? i think would prefer if they were all just public functions